pulsar-broker/src/test/java/org/apache/pulsar/client/impl/ReaderTest.java

```diff
@@ -785,4 +785,32 @@ public void testReaderListenerAcknowledgement()
         admin.topics().deletePartitionedTopic(partitionedTopic);
     }
 
+    @Test
+    public void testReaderReconnectedFromNextEntry() throws Exception {
```

Comment:
I tried to run this test without this fix many times.
But the test can get passed.

<img width="1649" alt="image" src="https://github.com/apache/pulsar/assets/12592133/197cdb56-49c8-44b8-af0e-9e273b145c6b">

Could you please check if the test can actually cover the issue that this PR try to fix?

Reply:
This PR will not affect the behavior of the reader. This test is just to cover the reader test.

Reply:
Ok, I see.